### ISSUE TITLE
Can't input after starting Fenrir

### ISSUE THREAD

**Issue by smythp** [chrys87/fenrir#60]
Hi, I'mn attempting to install Fenrir. I've gone through an installed dependencies, run the pulseaudio script as normal user and root, and run the install script. When I check dependenceis using the script, I get these:

```
python3-brlapi: FAIL
python3-speechd: FAIL
python3-espeak: FAIL
```

I'm on Ubuntu and python3-speechd is version 0.9.1-4. python3-espeak is 0.5-2. I'm not using braille.

When I start fenrir, I get the starting sound, but then can enter no input and hear nothing until cancelling the process with Control-c, when I get the exit sound. When I disable sound effects, I get a "start fenrir" and "quit fenrir" message. I'm using gnome-terminal on Ubuntu 20 LTS. Any guidance from here on getting fenrir working would be appreciated .

**Comment by smythp** [comment]
Is this project still active? What are folks using for a screen reader in 2024?

**Comment by stormdragon2976** [comment]
Yes, still active, used by lots. It is primary CLI screen reader for a
few distros.

**Comment by smythp** [comment]
OK, might try getting it working again, I've been using tdsr.

**Comment by chrys87** [comment]
Sorry for the late answer, i missed that somehow. how do you start fenrir exactly? Using systemd? What mode? What settings? Can you paste a debug out? 

sudo fenrir -d

should give you a file in /tmp/fenrirscreenreader

**Comment by smythp** [comment]
Here's the dependency check:

```
checking dependencys...

fenrir core:
python3-daemonize: FAIL
pyenchant: FAIL
--------------------
screen driver
dummyDriver (screen): OK
vcsaDriver
python3-dbus: OK
VCSA Device: OK

ptyDriver
pyte: FAIL
--------------------
braille driver
dummyDriver (braille): OK
debugDriver (braille): OK
brlapiDriver
python3-brlapi: FAIL
--------------------
input driver
dummyDriver (input): OK
debugDriver (input): OK
evdevDriver
python3-evdev: FAIL
python3-pyudev: FAIL

ptyDriver
pyte: FAIL
--------------------
sound driver
dummyDriver (sound): OK
debugDriver (sound): OK
genericDriver (uses sox by default)
sox: OK

gstreamerDriver
gi: OK
gi GLib: OK
gi Gst: OK
--------------------
speech driver
dummyDriver (speech): OK
debugDriver (speech): OK
speechdDriver
python3-speechd: FAIL

espeakDriver
python3-espeak: FAIL
genericDriver (uses espeak-ng by default)
espeak-ng: OK
====================
Default Setup: FAIL
Unavailable Default Modules:
FenrirCore
evdevDriver
you may need to install the missing dependencys for the modules above or reconfigure fenrir to not use them

Available Modules:
dummyDriver (screen)
vcsaDriver
dummyDriver (braille)
debugDriver (braille)
dummyDriver (input)
debugDriver (input)
dummyDriver (sound)
debugDriver (sound)
genericDriver (sound)
gstreamerDriver
dummyDriver (speech)
debugDriver (speech)
genericDriver (speech)
```

My OS:

```
NAME="Linux Mint"
VERSION="20.2 (Uma)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 20.2"
VERSION_ID="20.2"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=uma
UBUNTU_CODENAME=focal
```

Settings are the default.

Running sudo fenrir -d, I get a fenrirscreenreader-deamon.sock file in /tmp.

At startup I hear a sound play, at the same time I get espeak TTS with a message saying something is not installed I can't hear what that something is, though, it's very slightly cut off and the sound plays over it.

**Comment by stormdragon2976** [comment]
It looks like you are missing python-evdev. Can you search for and
install the evdev package and try again? Hopefully that will fix the
problem. Just to cover all bases, try launching with sudo fenrir for the
console, or sudo fenrir-daemon if you do not want it to consume the
terminal where it is launched.

**Comment by smythp** [comment]
```
python3-evdev is already the newest version (1.3.0+dfsg-1build1).
```

When I use `sudo fenrir`, I get the chord sound and a spoken message that something isn't installed, but the first word is garbled and I can't hear what it is (something short). There's no further interaction after that.

**Comment by chrys87** [comment]
well something seems to be broken with the python-evdev an d python-pyudev  packages.
the dependency check basically just import the modules as any python application would do. if this does not work for any reason, every other python application (fenrir as well) cannot import them.

what happens if you call 
`sudo python`
and add those two lines
`import pyudev
import evdev
`
do you get any sort of error?

**Comment by smythp** [comment]
Yeah, module not found for both. When I try to sudo pip install them, it says requirement already satisfied. It seems there's some disconnect between the root python executable and the root pip executable. Let me know if you have thoughts.
